alg/crc32c_sse42.c

```diff
@@ -15,17 +15,29 @@
 void
 CRC32C_Update_SSE42(uint32_t * state, const uint8_t * buf, size_t len)
 {
-	const size_t pre = ((uintptr_t)buf & 7) < len ? (uintptr_t)buf & 7 : len;
-	const size_t remaining = len - pre;
-	const size_t block_len = remaining - remaining % 8;
+	size_t pre_block;
+	size_t remaining_bytes;
+	size_t in_block;
 	size_t i = 0;
 
+	/*
+	 * Calculate how many bytes are before an aligned block.  Assume that
+	 * the (uintptr_t) value indicates the memory layout.
+	 */
+	pre_block = (uintptr_t)buf & 7;
```

Comment:
If `buf` is `0x10001`, what is `pre_block`?

And how many bytes do you need to process before you reach a multiple of `0x8`?